### ISSUE TITLE
database design for the app

### ISSUE THREAD

**Issue by epogrebnyak** [mini-kep/frontend-app#9]
Dataflow:
- there are macroeconomc data sources that interest us, some are static files and some are already the API backends 
- a parser is a directed pipeline of operations on the source
-  it can be implemented as is python script downloading/transformin data locally (if there is a lot to do, the source is dirty, etc) or an AWS lambda function (quick job, few transformations):
   -  a python script in github repo, maintained manually with stable URLto fetch data from, like *mini-kep*
   - AWS lambda that queries some other API and produces a json as the output
- parser produces some cleaner output from the source, say CSV/JSON file. This output may or may not be normalised. 
- then we have a 'raws' database and 'clean' database
- the 'raws' database contains identical copies of parser output. a daemon knows when to invoke parsers and how to store data in a 'raw' database
- after daemon collected data from the parsers into raw db it becomes the state of knwon information , further work is done on the raw database.
- there should be an algorithm for taking data from 'raw' to 'clean' database. I think clean database should be flat/normalised, each row representing a datapoint: ```<VARNAME>-<date>-<value>-<frequency>-<...>```
- the end user api request is fulfilled using the 'clean' database. For example, on ```GET varname='BRENT', frequency='d', start='2017-03-07'>``` we get something like 

```
2017-03-07,54.61
2017-03-08,53.3
2017-03-09,50.65
2017-03-10,50.63
2017-03-13,50.1
2017-03-14,49.6
2017-03-15,50.63
2017-03-16,50.56
2017-03-17,50.58
2017-03-20,50.67
```
- the end-user output format should be acceptable to pandas.read_json()  

In short that is:
```
1. ETL
source(s) -> parser(s) -> raw csv/json(s) -> raw db

2. Storage
raw db -> normalisation + any checks -> clean db

3. User query 
clean db on user request -> json -> pandas -> user  [-> ipython notebooks ]
```

**Comment by epogrebnyak** [comment]
Parser examlpes - all save output in local file. **mini-kep** is heavy and will be a local script. Other parser are thinner and can live inside AWS Lambda's.

#### github script: mini-kep
Produces output in <https://github.com/epogrebnyak/mini-kep/tree/master/data/processed/latest>

#### API parcer: cbr-usd (CB)
<https://github.com/ru-stat/parser-cbr-usd>

#### API parcer: yield curve (Treausry)
<https://github.com/epogrebnyak/ust>
result: <https://raw.githubusercontent.com/epogrebnyak/ust/master/ust.csv> (1.1 Mb)

#### API parcer: Brent (EIA)
<https://github.com/epogrebnyak/data-fx-oil/blob/master/eia.py>
result: <https://github.com/epogrebnyak/data-fx-oil/blob/master/brent_daily.txt>

**Comment by epogrebnyak** [comment]
Raws / clean database proposal rejected in favour of scheduler / clean database.
